roles/sap_vm_provision/tasks/platform_ansible/redhat_ocp_virt/execute_main.yml

```diff
@@ -0,0 +1,137 @@
+        loop_var: add_item
+
+# Cannot override any variables from extravars input, see https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence
+# Ensure no default value exists for any prompted variable before execution of Ansible Playbook
+
+    - name: Set fact to hold all inventory hosts in all groups
+      ansible.builtin.set_fact:
+        groups_merged_list: "{{ [ [ groups['hana_primary'] | default([]) ] , [ groups['hana_secondary'] | default([]) ] , [ groups['nwas_ascs'] | default([]) ] , [ groups['nwas_ers'] | default([]) ] , [ groups['nwas_pas'] | default([]) ] , [ groups['nwas_aas'] | default([]) ] ] | flatten | select() }}"
+
+    - name: Set Ansible Vars
+      register: __sap_vm_provision_task_ansible_vars_set
+      ansible.builtin.include_tasks:
+        file: common/set_ansible_vars.yml
+
+  rescue:
+    # This requires no_log set on each Ansible Task, and not set on the Ansible Task Block
+    # This requires an Ansible Task Block containing the Ansible Tasks for calling
+    # Infrastructure Platform APIs (via Ansible Modules)
+    - name: Show errors in task outputs
+      ansible.builtin.fail:
+        msg: "{{ lookup('ansible.builtin.vars', loop_item) }}"
+      loop:
+        - __sap_vm_provision_task_provision_host_all_run
+        - __sap_vm_provision_task_provision_host_single
+        - __sap_vm_provision_task_provision_host_single_info
+        - __sap_vm_provision_task_ansible_facts_host_disks_info
+        - __sap_vm_provision_task_provision_host_all_add
+      loop_control:
+        loop_var: loop_item
+        index_var: loop_item_index
+        label: "{{ 'Variable No. ' + (loop_item_index | string) }}"
+      when:
+        - lookup('ansible.builtin.vars', loop_item, default='') | length > 0
+        - not lookup('ansible.builtin.vars', loop_item, default='') is skipped
+        - lookup('ansible.builtin.vars', loop_item, default='') is failed
+
+- name: Write private ssh key to ansible_controller
+  delegate_to: "{{ sap_vm_provision_register_ansible_controller }}"
+  no_log: true
+  ansible.builtin.copy:
+    dest: "{{ sap_vm_provision_ssh_host_private_key_file_path }}"
+    content: "{{ sap_vm_provision_ssh_host_private_key_data }}"
+    mode: "0600"
+  when: sap_vm_provision_ocp_guest_ssh_auth_mechanism == "private_key_data"
+
+- name: Ansible Task block to execute on target inventory hosts
+  remote_user: "{{ sap_vm_provision_ocp_os_user }}"
+  become: true
+  become_user: root
+  delegate_to: "{{ inventory_hostname }}"
+  vars:
+    ansible_password: "{{ sap_vm_provision_ocp_os_user_password }}"
+    ansible_ssh_private_key_file: "{{ sap_vm_provision_ssh_host_private_key_file_path }}"
+    ansible_ssh_common_args: "-o ConnectTimeout=180 -o ControlMaster=auto -o ControlPersist=3600s -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ForwardX11=no -o ProxyJump={{ __sap_vm_provision_register_jumphost_user }}@{{ sap_vm_provision_jumphost }}"
```

Comment:
`-o StrictHostKeyChecking=no` Possible security issue?

Reply:
True, testing now without that option.

Reply:
Okay, it works without, will remove this option.